planner/core/plan_cache.go

```diff
@@ -167,8 +167,12 @@ func GetPlanFromSessionPlanCache(ctx context.Context, sctx sessionctx.Context,
 	}
 
 	if stmtCtx.UseCache { // for non-point plans
+		limitCountAndOffset, paramErr := ExtractLimitFromAst(stmt.PreparedAst.Stmt, sctx)
```

Comment:
Better to pass `limitCountAndOffset` into `generateNewPlan` so that we don't need to extract them again.